### PR TITLE
Feat/ Exhibition guide anchor links

### PR DESCRIPTION
**Pull request by melanierogan**
## Who is this for?

Exhibition guide users and editors https://github.com/wellcomecollection/wellcomecollection.org/issues/8359

## What is it doing for them?

Allowing us to have urls to link to individual stops within an exhibition guide page i.e. in audio (with or without description) and bsl we link to the title of the audio and video, in captions and transcripts we link to `standAlone` section titles, context titles, and tombstone titles. The url contains the id of the particular stop/section, the id outputs as `#the-title-of-the-stop-or-thing`.

Some examples:
**weco.org/guides/exhibitions/YvJ4UhAAAPgZztMl/audio-without-descriptions#weeping---joy-in-the-passions--humorously-delineated** brings you to that 'stop'

**weco.org/guides/exhibitions/YvJ4UhAAAPgZztMl/captions-and-transcripts#projection-and-fear** brings you to the projection and fear section 

**weco.org/guides/exhibitions/YvJ4UhAAAPgZztMl/captions-and-transcripts#joy-inside-our-tears** brings you to the tombstone title for that object in a stop

### CODE REVIEW

File: content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx [hunk 1]
```diff
@@ -202,7 +203,9 @@ const Stop: FC<{ stop: Stop; isFirstStop: boolean }> = ({
               }}
               v={{ size: 'l', properties: ['margin-bottom'] }}
             >
-              <StandaloneTitle>{standaloneTitle}</StandaloneTitle>
+              <StandaloneTitle id={dasherize(`${standaloneTitle}`)}>
```
Comment (alexwlchan) [inline content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx:206]: nit: I think you can do away with the interpolation here

```suggestion
              <StandaloneTitle id={dasherize(standaloneTitle)}>
```

File: content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx [hunk 2]
```diff
@@ -222,7 +225,9 @@ const Stop: FC<{ stop: Stop; isFirstStop: boolean }> = ({
         >
           <div className="flex flex--wrap container">
             <Tombstone>
-              <TombstoneTitle>{!hasContext && title}</TombstoneTitle>
+              <TombstoneTitle id={dasherize(`${title}`)}>
```
Comment (alexwlchan) [inline content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx:228]:
```suggestion
              <TombstoneTitle id={dasherize(title)}>
```

File: content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx [hunk 3]
```diff
@@ -238,14 +243,18 @@ const Stop: FC<{ stop: Stop; isFirstStop: boolean }> = ({
...
+                    {title}
+                  </StandaloneTitle>
                 </Space>
               )}
 
               {hasContext && (
                 <>
                   {!isFirstStop && title.length > 0 && (
-                    <ContextTitle>{title}</ContextTitle>
+                    <ContextTitle id={dasherize(`${title}`)}>
```
Comment (alexwlchan) [inline content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx:255]:
```suggestion
                    <ContextTitle id={dasherize(title)}>
```

**Review by alexwlchan — APPROVED** [review]
Looks good. ⚓ 

My only thought is whether titles are the best thing to use in the `id` attribute; do we know how long they might be? Some of the catalogue titles are monstrous; I hope exhibition titles won't be that long.

Is there anything like a "stop number" we could use for more predictably-sized URLs? I guess we have the `index` variable on stops, but not on the standalone titles?

File: content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx (hunk 3 above)
Comment (melanierogan) [reply content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx:255]: Danke!

**Comment by melanierogan** [comment]
> Looks good. ⚓
> 
> My only thought is whether titles are the best thing to use in the `id` attribute; do we know how long they might be? Some of the catalogue titles are monstrous; I hope exhibition titles won't be that long.
> 
> Is there anything like a "stop number" we could use for more predictably-sized URLs? I guess we have the `index` variable on stops, but not on the standalone titles?

I hear you on the long titles, I was basing the approach off what we have used for qr codes in guides previously e.g. https://wellcomecollection.org/pages/YOwMVBMAACMATaPW

I think as the anchor links are going to be used to create qr codes for specific areas, we want them to have recognisable ids to editors or creators of the titles, like @dannybirchall and his team.

File: content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx (hunk 1 above)
Comment (melanierogan) [reply content/webapp/components/ExhibitionCaptions/ExhibitionCaptions.tsx:206]: Ah i remember why i did this now, standaloneTItle is a `prismicT.RichTextField` so I'm ensuring I change it to a `string` to keep Typescript and the dasherize function happy. The other places I've placed dasherize work with `title` which is a `string`.

**Comment by dannybirchall** [comment]
++ to predictability over legibility in URLs

**Pull request merged**